core/tracing/hooks.go

```diff
@@ -232,8 +232,113 @@ const (
 	// account within the same tx (captured at end of tx).
 	// Note it doesn't account for a self-destruct which appoints itself as recipient.
 	BalanceDecreaseSelfdestructBurn BalanceChangeReason = 14
+
+	// Arbitrum specific
+	BalanceIncreaseDeposit           BalanceChangeReason = 15
+	BalanceDecreaseWithdrawToL1      BalanceChangeReason = 16
+	BalanceIncreaseL1PosterFee       BalanceChangeReason = 17
+	BalanceIncreaseInfraFee          BalanceChangeReason = 18
+	BalanceIncreaseNetworkFee        BalanceChangeReason = 19
+	BalanceIncreaseRetryTxPrepaid    BalanceChangeReason = 20
+	BalanceDecreaseRetryTxUndoRefund BalanceChangeReason = 21
+
+	BalanceChangeTransferRetryTxToEscrow   BalanceChangeReason = 22
+	BalanceChangeTransferRetryTxFromEscrow BalanceChangeReason = 23
+
+	BalanceChangeTransferRetryableToFeeRefundAddr BalanceChangeReason = 24
+	BalanceChangeTransferRetryableToEscrow        BalanceChangeReason = 25
+
+	BalanceChangeTransferRetryableToInfra     BalanceChangeReason = 26
+	BalanceChangeTransferRetryableFromInfra   BalanceChangeReason = 27
+	BalanceChangeTransferRetryableToNetwork   BalanceChangeReason = 28
+	BalanceChangeTransferRetryableFromNetwork BalanceChangeReason = 29
+
+	BalanceChangeTransferBatchposterReward BalanceChangeReason = 30
+	BalanceChangeTransferBatchposterRefund BalanceChangeReason = 31
+
+	// Stylus
+	BalanceChangeTransferActivationFee       BalanceChangeReason = 32
+	BalanceChangeTransferActivationReimburse BalanceChangeReason = 33
 )
 
+func (b BalanceChangeReason) String(prev, new *big.Int) string {
```

Comment:
The thing is geth itself doesn't use `BalanceChangeReason` type for anything meaningful (as of 1.14.0) i.e in any of the tracing code, but gives us the freedom to decide how and where we use it. 

They aren't formatted after anything standard but gives the reason for the balance change itself, which I thought we could use in tracing, let me know if I got it wrong.

For example geth's current master uses `BalanceChangeReason` this way https://github.com/ethereum/go-ethereum/blob/6d3d252a5ece0687dcaaf197e0bfc03466161ff1/eth/tracers/live/supply.go#L191-L208

Reply:
I am not sure, but I guess someone is already parsing the exiting short srings.
If they are - this change would hurt them, and I rather avoid it.
We could still have a "String" function but have it return the old string to avoid compatibility errors.

Reply:
Changed the `String()` to be backwards compatible